indico/modules/rb/client/js/common/rooms/edit/RoomEditModal.jsx

```diff
@@ -0,0 +1,435 @@
+                    type="button"
+                    icon="undo"
+                    size="mini"
+                    color="red"
+                    content={Translate.string('Reset form')}
+                    onClick={() => formProps.form.reset()}
+                  />
+                </Message>
+              );
+            }}
+          </FormSpy>
+          <Grid columns="equal">
+            {!isNewRoom && (
+              <Grid.Column width={4}>
+                <RoomPhoto roomId={roomId} hasPhoto={roomDetails && roomDetails.has_photo} />
+              </Grid.Column>
+            )}
+            <Grid.Column>
+              <Form id="room-form" onSubmit={handleFormSubmit}>
+                <Tab
+                  renderActiveOnly={false}
+                  panes={tabPanes}
+                  onTabChange={(__, {activeIndex}) => {
+                    setActiveTab(tabPanes[activeIndex].key);
+                  }}
+                />
+              </Form>
+            </Grid.Column>
+          </Grid>
+          <FormSpy
+            subscription={{errors: true, submitErrors: true}}
+            onChange={({errors, submitErrors}) => {
+              setTabsWithError(
+                Object.keys({...errors, ...submitErrors}).map(err => {
+                  const pane = tabPanes.find(t => t.fields && t.fields.includes(err));
+                  return pane && pane.key;
+                })
+              );
+            }}
+          />
+        </Modal.Content>
+        <Modal.Actions>
+          <FormSpy subscription={{submitSucceeded: true}}>
+            {({submitSucceeded}) => (
+              <Button onClick={onClose}>
+                {submitSucceeded ? <Translate>Close</Translate> : <Translate>Cancel</Translate>}
+              </Button>
+            )}
+          </FormSpy>
+          <Button
+            type="submit"
+            form="room-form"
+            disabled={pristine || submitting || hasValidationErrors}
+            loading={submitting}
+            primary
+          >
+            <Translate>Save</Translate>
+          </Button>
+        </Modal.Actions>
+      </Modal>
+    );
+  };
+
+  const renderForm = () => {
+    if (newRoomId) {
+      return <RoomEditModal roomId={newRoomId} onClose={onClose} afterCreation />;
+    }
+    if (!roomDetails || !roomAvailability || !roomAttributes) {
+      // In case of error, nothing is returned
+      return null;
+    }
+
+    return (
+      <FinalForm
+        onSubmit={handleSubmit}
+        initialValues={{
+          ...roomDetails,
+          ...roomAvailability,
+          attributes: roomAttributes,
+        }}
+        initialValuesEqual={_.isEqual}
+        render={renderModal}
+        subscription={{
+          submitting: true,
+          hasValidationErrors: true,
+          pristine: true,
+          dirty: true,
+        }}
+        mutators={{...arrayMutators}}
+        validate={formValidation}
+      />
+    );
+  };
+
+  return renderForm();
```

Comment:
What's the reason for having this inside a function? I think removing the `const renderForm = () => { ... }` around it would still keep the exact same behavior

Reply:
I don't see any reason either. Initially, it was because the function was going to be placed somewhere else. I'll move it outside.